### DISCUSSION TITLE
"BIO_new_dgram_sctp" Not Recognized During Linker Stage

### DISCUSSION THREAD

**Discussion by lakshya-chopra** [openssl/openssl#24948] [Q&A] [answered]
I was following the **DTLS over SCTP** examples from [this repository](https://github.com/nplab/DTLS-Examples/tree/master). The code makes use OpenSSL BIO specifically for SCTP semantics, i.e., **BIO_new_dgram_sctp**. I followed the documentation, and installed the required packages (`libsctp-dev`), and configured OpenSSL for SCTP with the following commands:

```sh
./config sctp --prefix=/usr/local/
make
make install
```
For reference, I'm using the openssl version `3.3.1`.

So, the installation did succeed, and the support for sctp was enabled as well (can be verified by running `openssl s_client -help` and checking if sctp's there). 
![image](https://github.com/user-attachments/assets/ef2c5eef-2218-4388-9e5c-619b338af32d)

However, when I tried to compile the files from the provided repository, I kept getting the linker error:
```
undefined reference to `BIO_new_dgram_sctp'
collect2: error: ld returned 1 exit status
```
I have updated the linker path and provided the correct headers, but I’m still facing the issue. Am i missing something? Any insights regarding this would be appreciated.

**Comment by t8m** [comment, marked as answer]
Most likely your application is being linked to some other libcrypto.so than the one you've built and installed. Try to look into /usr/lib and /usr/lib64 if there isn't another libcrypto.so.
